### DISCUSSION TITLE
First submit does not trigger dirty

### DISCUSSION THREAD

**Discussion by webartoli** [react-hook-form/react-hook-form#3951] [Q&A] [answered]
Hi there,
  I have to send to backend server only changed fields of my form.

So i need to find dirty status and dirty fields of my form.
In the example below thw first shoot seems broken.
https://codesandbox.io/s/react-hook-form-controller-template-ok-5iy6b?file=/src/index.js

To Reproduce:
- Change Any field
- Send form (form is not dirty)
- Change Any field (i.e. the same field)
- Send form (form is dirty)

What I'm missing ?

**Comment by webartoli** [comment]
Additional (strange) details about the observed behaviour: adding a piece of dom that trigger form.isDirty this seems works as expected.

``` tsx
<form onSubmit={handleSubmit(onSubmit, reset)}>
   { /* ... */}
   { formState.isDirty && <p>Dirty</p>}

   <input type="submit" />
</form>
```

**Comment by borm** [comment, marked as answer]
@webartoli u need use 
```
const { isDirty, dirtyFields } = formState;
  const onSubmit = (data) => {
    console.log(isDirty, dirtyFields);
    console.log(data);
    reset(data);
  };
```

https://codesandbox.io/s/react-hook-form-controller-template-date-not-dirty-forked-ls4uo

https://react-hook-form.com/api#formState
// Read the formState before render to subscribe the form state through Proxy

**Reply by bluebill1049** [reply]
Thanks @borm for helping.
